### DISCUSSION TITLE
Set page number for a document page

### DISCUSSION THREAD

**Discussion by Thecode764** [squidfunk/mkdocs-material#7306] [Q&A]
hi. i am new to mkdocs and mkdocs-material. hoe to set page number in the sidebar?

**Comment by kamilkrzyskow** [comment]
Hello @Thecode764,
what do you mean by page number? You mean an increasing count number that keeps track of pages, or do you mean something like pagination? Both would require some/a lot of customization on your part.

If you're asking about pagination, then this is rather individualized, maybe this thread could give you some ideas how to tackle the issue:
- https://github.com/squidfunk/mkdocs-material/discussions/6330

As for a page count, please have in mind that pages aren't processed via their modification/creation date, I think they're processed alphabetically, so an automatic increasing count could render differently based on the amount of added files to your docs.
This would be done via a hooks in Python if you still want to make it automatic: 
- https://www.mkdocs.org/user-guide/configuration/#hooks

I would recommend using a more manual approach of adding a number to the front matter meta at the top of your files.
```yml
---
count_number: 1
---
```
Then you would modify the templates, either `nav` or `toc` to make them use the special `page.meta.count_number` and make it appear in the side panel.
- https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme
- https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/toc.html
- https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/nav.html

**Reply by kamilkrzyskow** [reply]
@Thecode764 
How was the issue resolved in the end? Can you please share your solution or at least the general gist of it?
